react/components/EXPERIMENTAL_Table/Toolbar/ButtonColumns.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useMemo, FC } from 'react'
+
+import Toggle from '../../../Toggle'
+import IconColumns from '../../icon/Columns/index'
+import Menu from './Menu/index'
+
+import { ICON_SIZE, COLUMNS_BOX, NAMESPACES } from '../constants'
+import useTableContext from '../hooks/useTableContext'
+
+export type ButtonColumnsProps = {
+  label: string
+  showAllLabel: string
+  hideAllLabel: string
+  alignMenu: Alignment
+  disabled: boolean
+}
+
+const ButtonColumns: FC<ButtonColumnsProps> = ({
+  label,
+  showAllLabel,
+  hideAllLabel,
+  alignMenu,
+  disabled,
+}) => {
+  const {
+    hiddenColumns,
+    columns,
+    hideAllColumns,
+    showAllColumns,
+    toggleColumn,
+  } = useTableContext()
+
+  const calculateFieldsBoxHeight = () => {
+    const estimate = Object.keys(columns).length * COLUMNS_BOX.ITEM_HEIGHT
+    return estimate > COLUMNS_BOX.MAX_HEIGHT ? COLUMNS_BOX.MAX_HEIGHT : estimate
```

Comment:
Instead of do this ternary comparison it is enough to return the minimum between both values. Something like `Math.min(estimate, COLUMNS_BOX.MAX_HEIGHT)` or even `Math.min(Object.keys(columns).length * COLUMNS_BOX.ITEM_HEIGHT, COLUMNS_BOX.MAX_HEIGHT)`